WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -128,13 +110,20 @@ def update
   end
 
   private def user_params
-    user_params = params.require(:user).permit(*current_user.editable_fields_of_user(user_to_edit))
+    permitted_params = current_user.editable_fields_of_user(user_to_edit).to_a
+    permitted_params.delete :preferred_event_ids
+    permitted_params.push(preferred_event_ids: Event.all_official.map(&:id))
+
+    user_params = params.require(:user).permit(permitted_params)
     if user_params.key?(:delegate_status) && !User.delegate_status_allows_senior_delegate(user_params[:delegate_status])
       user_params["senior_delegate_id"] = nil
     end
     if user_params.key?(:wca_id)
       user_params[:wca_id] = user_params[:wca_id].upcase
     end
+    if user_params.key?(:preferred_event_ids)
+      user_params[:preferred_event_ids] = user_params[:preferred_event_ids].select { |_k, v| v == "1" }.keys
```

Comment:
I think we can leave it as it is in this case because of [this](https://github.com/cubing/worldcubeassociation.org/blob/master/WcaOnRails/app/inputs/events_picker_input.rb#L9-L10) =)